### ISSUE TITLE
2D point clouds?

### ISSUE THREAD

**Issue by darribas** [timkittel/alpha-shapes#1]
Hello,

I first wanted to say thank you very much for this little library, it does what it's supposed to and it's very fast! Second I wanted to ask about the possibility of computing alpha shapes for two dimensional point clouds. I've noticed the library currently only accepts three dimensions. I need this functionality so I was going to have a try at modifying your code to adapt it. I know Python but not an expert in computational geometry, would you have any leads or suggestions that could help me? I'd be happy to collaborate or contribute back the resulting code if you want.

Thank you very much in advance.

**Comment by timziebart** [comment]
Hi,
thanks a lot for your compliments and interest, I am happy you like it (: 
I think it would be little work to adapt it to 2d point clouds. If you are not in a hurry, I would give you some pointers next Monday. Does that sounds good to you?
Cheers, Tim

**Comment by darribas** [comment]
Thanks very much for getting back! Yes, no rush at all. Any sort of information will be helpful. After looking at the main file, it looks like it'd be transforming tetrahedrons into triangles, reducing dimensionality then to areas rather than volumes, but there are a few bits that I could do with some help. Maybe a better approach not to waste your time is for me to have a go at it and issue a PR that you can comment on?

Thanks again,

]d[

**Comment by timziebart** [comment]
That sounds about right. [Here is a page](https://www.mathopenref.com/trianglecircumcircle.html) with an equation for the radius of the circumcircle of a triangle.
Yes, maybe a PR would be a good way. I guess the basic structure would be straightforward and I could help with getting the speed via numba (:
Ah, and I realized I haven't added a license yet, so I would simply go for MIT, as it's scipy compatible. I hope you agree as you contribution will be published under it, too.

**Comment by darribas** [comment]
Yes, a permissible scipy/pydata compatible would be fantastic. If we get around it later, we could document a bit and put up on PIP :-) 

I'll issue a PR and we can discuss issues on it. Then you can either merge or we discuss how to integrate the two before merging.

**Comment by timziebart** [comment]
Yep, that would be great. Actually, I am rather amazed that you found and used it without at least a readme (:

looking forward to see your code

**Comment by timziebart** [comment]
I added the BSD-3 clause as it is the [scipy license](https://www.scipy.org/scipylib/license.html).

**Comment by darribas** [comment]
I have a working implementation for 2D clouds based off your 3D. Would be happy to contribute back here but not sure what the best API would be. Also, considering contributing it to [PySAL](https://github.com/pysal/pysal) as it would be covered by CI, released to a larger audience, etc. I've started a ticket ([pysal#1036](https://github.com/pysal/pysal/issues/1036)) to figure out what the best way to do it would be. We might also consider integrating your 3D code if you were interested. What do you think?

**Comment by timziebart** [comment]
Yep, I would totally agree contributing to PySAL, would be happy if the code find application. 
For the Code here, the best API would probably be to simple check for the dimension of the number of points when calling `get_alpha_shape` as it is simply[ this line](https://github.com/timkittel/alpha-shapes/blob/master/alpha_shapes/alpha_shapes.py#L86) (and the functions behind it) that would be different for 2D. The rest should be the same, right?

Concerning PySAL, I will write right in the Issue you referenced.

**Comment by darribas** [comment]
Excellent! Should we focus on one place rather than duplicating? My call would obviously go for PySAL as it'll probably ensure more people use it. Given the license, it can be in as many places as we want, but just thinking of what would make most sense.
